router/internal/router/config.go

```diff
@@ -248,6 +253,12 @@ func (config *Config) loadListener(rootConfigMap map[interface{}]interface{}) er
 					return errors.New("required value [listeners.edge.options.advertise] was not a string or was not found")
 				}
 
+				parts := strings.Split(advertise, ":")
+				if parts[1] != edgeListenPort {
+					msg := fmt.Sprintf("port in [listeners.edge.options.advertise] must equal port in [listeners.edge.address] but did not. Got [%s] [%s]", parts[1], edgeListenPort)
```

Comment:
The ports don't have to be equal. It is possible the advertised address is some proxy that listens on a different port.

Reply:
Fair enough, but I will emit a WARN log msg here to help call out cases where someone might have typo'd (like I have done).